packages/markdown-support/src/index.ts

```diff
@@ -11,52 +11,43 @@ import markdownToHtml from 'remark-rehype';
 // import smartypants from '@silvenon/remark-smartypants';
 import rehypeStringify from 'rehype-stringify';
 
-export interface MarkdownRenderingOptions extends Partial<AstroMarkdownOptions> {
-  $?: {
-    scopedClassName: string | null;
-  };
-  mode: 'md' | 'astro-md';
-}
+export { AstroMarkdownOptions, MarkdownRenderingOptions };
 
 /** Internal utility for rendering a full markdown file and extracting Frontmatter data */
 export async function renderMarkdownWithFrontmatter(contents: string, opts?: MarkdownRenderingOptions | null) {
   // Dynamic import to ensure that "gray-matter" isn't built by Snowpack
   const { default: matter } = await import('gray-matter');
   const { data: frontmatter, content } = matter(contents);
-  const value = await renderMarkdown(content, { ...opts, mode: 'md' });
+  const value = await renderMarkdown(content, opts);
   return { ...value, frontmatter };
 }
 
 /** Shared utility for rendering markdown */
 export async function renderMarkdown(content: string, opts?: MarkdownRenderingOptions | null) {
-  const { $: { scopedClassName = null } = {}, mode = 'astro-md', footnotes: useFootnotes = true, gfm: useGfm = true } = opts ?? {};
+  const { $: { scopedClassName = null } = {}, remarkPlugins = [], rehypePlugins = [] } = opts ?? {};
   const { headers, rehypeCollectHeaders } = createCollectHeaders();
 
-  let parser = unified().use(markdown).use(remarkCodeBlock());
+  let parser = unified().use(markdown);
 
   if (scopedClassName) {
     parser = parser.use(scopedStyles(scopedClassName));
   }
+  const loadedRemarkPlugins = await Promise.all(remarkPlugins);
+  const loadedRehypePlugins = await Promise.all(rehypePlugins);
 
-  if (useGfm) {
-    const { default: gfm } = await import('remark-gfm');
-    parser = parser.use(gfm);
-  }
+  loadedRemarkPlugins.forEach((p) => {
+    parser = parser.use(p.default);
+  });
 
-  if (useFootnotes) {
-    const { default: footnotes } = await import('remark-footnotes');
-    parser = parser.use(footnotes);
-  }
+  parser.use(markdownToHtml, { allowDangerousHtml: true, passThrough: ['raw'] });
+
+  loadedRehypePlugins.forEach((p) => {
```

Comment:
It seems like remark and rehype plugins are instantiated the same way, in which case is there a reason to differentiate them in the config?

Reply:
Yes, it works as I said above.

Reply:
Yep, some plugins require both a `remark` plugin for Markdown and a `rehype` plugin for the HTML, and we need to split them out because of when each one must run.